### PR TITLE
feat!: Add support for Wynncraft 2.1 items

### PR DESCRIPTION
**Pull request by kristofbolyai**
The API is still in beta and has loads of issues, so this will take a while..

### CODE REVIEW

**Comment by noahmfoster** [comment]
Wynntils does not recognize my Hanafubuki as a weapon.

**Comment by kristofbolyai** [comment]
> Wynntils does not recognize my Hanafubuki as a weapon. 

What do you think this PR is for?

File: common/src/main/java/com/wynntils/models/gear/GearInfoRegistry.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ public GearInfo deserialize(JsonElement jsonElement, Type jsonType, JsonDeserial
                 throw new RuntimeException("Invalid Wynncraft data: item has no gear type");
             }
 
-            GearTier tier = GearTier.fromString(json.get("tier").getAsString());
+            GearTier tier = GearTier.fromString(json.get("rarity").getAsString());
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/gear/GearInfoRegistry.java:138]: Do this reflect a general change in terminology from Wynncraft's side, that we should follow up in our code ("tier" -> "rarity")?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/gear/GearInfoRegistry.java:138]: I think this is just a semi random rename. They did change to using "armour" consistently, though.

**Comment by kristofbolyai** [comment]
This should be fully working now (as of V3.3 API dev-beta). Please review this, but don't merge it.

File: common/src/main/java/com/wynntils/models/ingredients/IngredientInfoRegistry.java [hunk 2]
```diff
@@ -111,7 +111,11 @@ public IngredientInfo deserialize(JsonElement jsonElement, Type jsonType, JsonDe
 
             Optional<String> internalNameOpt = Optional.ofNullable(internalName);
 
-            int tier = JsonUtils.getNullableJsonInt(json, "tier");
+            // The tier field was renamed to rarity in API v3.3, but it really does not make sense, so
+            // I'll keep parsing both tier and rarity for now (as they may change it back)
+            int tier = json.has("tier")
+                    ? JsonUtils.getNullableJsonInt(json, "tier")
+                    : JsonUtils.getNullableJsonInt(json, "rarity");
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/models/ingredients/IngredientInfoRegistry.java]: They mentioned changing this to back to tier so I don't think you need this anymore

**Comment by ryanzhoudev** [comment]
Seems fine in game. I tried with completely new 2.1 items and also a couple of my old stuff with new IDs.

File: common/src/main/java/com/wynntils/models/ingredients/IngredientInfoRegistry.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/models/ingredients/IngredientInfoRegistry.java]: Let's keep both for now; if they stabilize we can clean this up later. 

I've come to deeply mistrust the Wynncraft API. :-( They have had a hard time getting their act together on that...

**Review by magicus — APPROVED** [review]
Changes look sane. I have not looked at the new format so can't really judge the details, but if it works I guess it's fine.

**Pull request merged**